test/tests/functional/pbs_equiv_classes.py

```diff
@@ -1971,3 +1971,59 @@ def test_limit_res(self):
         # one for no foores
         self.scheduler.log_match("Number of job equivalence classes: 3",
                                  starttime=self.t)
+
+    def change_res(self, name, total, node_num, attribs):
+        """
+        Change the value of memory on one of the node
```

Comment:
@agrawalravi90 this is a supported way to changing attributes on the vnodes that create_vnodes() interface is trying to create. I can change it to calling create_vnodes() twice in a row but I think there is nothing wrong in providing a callback that can do it for the interface.
I'd like to retain this change as is.

Reply:
please do remove total and name arguments from the function, we can add them later if there's a need for them.

Reply:
I don't think that is possible. Like I said this is a callback function. Its signature is defined by PTL framework's create_vnode() method.

Reply:
ok, this just looks very ugly to me. The function is taking up 17 lines, you can accomplish the same with just 1 line of an additional create_vnodes call instead. I think it's good that we have the capability in create_vnodes to accept a callback function, but if there's an easier, cleaner way to do the same then i think we should do that instead of using the callback function.

Reply:
I strongly disagree with your comment. Callback function is 5 lines of code but what you are saying will be one line of code which will internally create and add vnodedef file into mom. Your approach is slow and inefficient. I’m not changing the code to do that. 
If you think callbacks are not easy or clean to use then we can have a wider discussion on forum about it.